### PR TITLE
Add Tests for Recipe Creation and Enhance RecipesViewModel Testing

### PR DESCRIPTION
**Pull request by 2Fick**
## Description
- **What has been changed?**
  - Enhanced the `RecipesViewModel` tests by adding assertions to check the loading state and ensuring that no recipes are set on failure.
  - Added unit tests for the creation of `Recipe` instances with both valid and nullable properties.
  - Refactored the `RecipesViewModel` to remove the initial fetch of random recipes in the `init` block, facilitating easier testing.
  ```kotlin
  init {
    // Removed the fetchRandomRecipes() call from the init block for testing purposes
  }
  ```
Please do not hesitate to make any suggestions for the init block if you have any.

- **Why was this change made?**
  These changes were made to improve the test coverage and reliability of the `RecipesViewModel` by ensuring that the loading state is properly managed and that the ViewModel starts with an empty state. This refactor also makes the ViewModel easier to test by removing side effects from the initialization process.

## Checklist

- [x] **Tests added**:
  - `create Recipe with nullable properties` (in `RecipeTest.kt`): This test ensures that `Recipe` instances can be created with nullable properties, verifying the class's behavior with optional fields.
  - `fetchRandomRecipesHandlesFailure` (in `RecipesViewModelTest.kt`): This test simulates a failure in fetching random recipes and checks that the loading state is set to false and no recipes are set.
  - `initialStateIsCorrect` (in `RecipesViewModelTest.kt`): This test verifies that the initial state of the `RecipesViewModel` is correct, ensuring that the recipes list is empty and the loading state is false.

- [x] **Tests modified**:
  - `fetchRandomRecipesUpdatesState` (in `RecipesViewModelTest.kt`): Added an assertion to check that the loading state is set to false after fetching recipes.
    ```kotlin
    assertThat(recipesViewModel.loading.value, `is`(false)) // Check loading is false after fetching
    ```

- [ ] Documentation updated.
- [x] All CI checks passed.
- [x] Linked issue/feature: #107

### CODE REVIEW

File: app/src/main/java/com/android/sample/model/recipe/RecipesViewModel.kt [hunk 1]
```diff
@@ -24,8 +24,7 @@ class RecipesViewModel(private val repository: RecipeRepository) : ViewModel() {
     get() = _loading
 
   init {
-    // Fetch a default number of random recipes when ViewModel is created
-    fetchRandomRecipes(5) // Fetch 5 random recipes as an example
+    // Removed the fetchRandomRecipes() call from the init block for testing purposes
   }
 
   /**
```
Comment (AdriB02) [inline app/src/main/java/com/android/sample/model/recipe/RecipesViewModel.kt:30]: Please ensure that fetchRandomRecipes properly handles cases based on the value of numberOfRecipes (e.g., when it's less than 0, equal to 0, or greater than 0). Consider whether this logic should be implemented in the RecipeRepository.

File: app/src/test/java/com/android/sample/model/recipe/RecipeTest.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    // Assert
+    assertThat(recipe.idMeal, `is`(idMeal))
+    assertThat(recipe.strMeal, `is`(strMeal))
+    assertThat(recipe.strCategory, `is`(nullValue())) // Use nullValue() matcher
+    assertThat(recipe.strArea, `is`(nullValue())) // Use nullValue() matcher
+    assertThat(recipe.strInstructions, `is`(strInstructions))
+    assertThat(recipe.strMealThumbUrl, `is`(strMealThumbUrl))
+    assertThat(recipe.ingredientsAndMeasurements, `is`(ingredientsAndMeasurements))
+  }
```
Comment (AdriB02) [inline app/src/test/java/com/android/sample/model/recipe/RecipeTest.kt:71]: You might want to handle the edge case where the ingredientsAndMeasurements list is empty

**Review by AdriB02 — COMMENTED** [review]
The test classes and `RecipesViewModel` are well-implemented, with effective use of `StateFlow` for managing recipe and loading states. The tests thoroughly cover both success and failure cases, ensuring robustness. Comments throughout the code are clear and helpful, improving readability. All essential features required for the swipe activity are present, and the overall structure is solid. Great work !

File: app/src/main/java/com/android/sample/model/recipe/RecipesViewModel.kt (hunk 1 above)
Comment (2Fick) [reply app/src/main/java/com/android/sample/model/recipe/RecipesViewModel.kt:35]: The fetchRandomRecipes function includes now a check to ensure numberOfRecipes is at least 1.

File: app/src/test/java/com/android/sample/model/recipe/RecipeTest.kt (hunk 2 above)
Comment (2Fick) [reply app/src/test/java/com/android/sample/model/recipe/RecipeTest.kt:72]: The Recipe class has been updated to validate that ingredientsAndMeasurements is not empty.

File: app/src/main/java/com/android/sample/model/recipe/RecipesViewModel.kt [hunk 3]
```diff
@@ -24,8 +24,7 @@ class RecipesViewModel(private val repository: RecipeRepository) : ViewModel() {
     get() = _loading
```
[the remark this answers is not in the document]
Comment (2Fick) [reply app/src/main/java/com/android/sample/model/recipe/RecipesViewModel.kt:25]: A StateFlow for the current selected recipe has been added, along with functions to update and clear the current recipe.

**Comment by AdriB02** [comment]
The changes look great! I see you’ve added a new `StateFlow` to represent the show recipe and functions to manage it, which is a solid improvement. The tests are well-written, and I appreciate the update to the `Recipe` class to ensure ? `ingredientsAndMeasurements` is not empty. Also, the check in `fetchRandomRecipes` to validate that `numberOfRecipes` is at least 1 is a nice addition. The code looks great and is ready to be merged!

**Review by AdriB02 — APPROVED** [review]
The changes look great! I see you’ve added a new `StateFlow` to represent the show recipe and functions to manage it, which is a solid improvement. The tests are well-written, and I appreciate the update to the `Recipe` class to ensure ? `ingredientsAndMeasurements` is not empty. Also, the check in `fetchRandomRecipes` to validate that `numberOfRecipes` is at least 1 is a nice addition. The code looks great and is ready to be merged!
